kernel/arch/dreamcast/hardware/modem/mintr.c

```diff
@@ -743,40 +760,17 @@ void modemIntResetControlCode(void) {
     modemCallbackCode = NULL;
 }
 
-/* Timeout timer interrupt code. It uses TMU1 and can be setup to either set
-   a flag to a non zero value, call a function, or both when a timer interrupt
-   is generated until it's stopped. */
-
-static void modemIntrTimeoutCallback(irq_t source, irq_context_t *context,
-                                     void *data) {
-    (void)source;
-    (void)context;
-    (void)data;
-
-    if(modemTimeoutCallbackFlag != NULL)
-        *modemTimeoutCallbackFlag = 1;
-
-    if(modemTimeoutCallbackCode != NULL)
-        modemTimeoutCallbackCode();
-}
-
 void modemIntSetupTimeoutTimer(int bps, unsigned char *callbackFlag,
                                void (*callbackCode)(void)) {
+    wdt_timer_stop(wdt);
```

Comment:
```suggestion
    (void)bps;
    
    wdt_timer_stop(wdt);
```

Reply:
Even with my changes above I couldnt get it. to work. I tried running the PPP example in examples/dreamcast/modem.  Using a DreamPi I was able to connect with the master branch but not your branch.  The output from the result:

```
net_dev_init: detected 0 usable network device(s)

Dialing connection
Couldn't dial a connection (-4)
```

Reply:
> 1000/bps since the code you replaced `timer_prime(TMU1, bps, 1);` where bps is number of ticks per second.

Honestly, I don't really understand the original timer code. If I read correctly, it sets TMU1 to "bps" ticks per second, but also sets the timer value to "bps" which means that the interrupt will always happen in one second, independently of the "bps" value; and that's why I configure for a timeout of 1000 ms.

Reply:
Honestly, I really dont understand the original timer code either.  I was working off comments in the modem code that may no longer be true.  I'll get more acquainted with the timer code tonight/tomorrow and see what your branch is doing differently.